x/gamm/pool-models/balancer/pool.go

```diff
@@ -663,6 +663,19 @@ func (p *Pool) JoinPool(ctx sdk.Context, tokensIn sdk.Coins, swapFee sdk.Dec) (n
 	return numShares, nil
 }
 
+// JoinPoolNoSwap calculates the number of shares needed for an all-asset join given tokensIn with swapFee applied.
+// It updates the liquidity if the pool is joined successfully. If not, returns error.
+func (p *Pool) JoinPoolNoSwap(ctx sdk.Context, tokensIn sdk.Coins, swapFee sdk.Dec) (numShares sdk.Int, err error) {
+	numShares, newLiquidity, _, err := p.CalcJoinPoolNoSwapShares(ctx, tokensIn, swapFee)
+	if err != nil {
+		return sdk.Int{}, err
+	}
+
+	// update pool with the calculated share and liquidity needed to join pool
+	p.IncreaseLiquidity(numShares, newLiquidity)
+	return numShares, nil
+}
+
 // CalcJoinPoolShares calculates the number of shares created to join pool with the provided amount of `tokenIn`.
```

Comment:
So now this method is basically not called in the codebase anywhere, right?

Reply:
`CalcJoinPoolShares`? Do you mean calls outside of the module? Otherwise it's called in `JoinPool`

Reply:
But JoinPool is no longer called in the state machien right?

Reply:
I might be missing your point here but I believe `JoinPool` is still called in `JoinSwapExactAmountIn`: https://github.com/osmosis-labs/osmosis/blob/16e3e4bdaa29006387a250121e9939e5d1d013df/x/gamm/keeper/pool_service.go#L272